docs/source/reference/api-server/examples/example-deploy-gcp-cloud-sql.rst

```diff
@@ -0,0 +1,276 @@
+
+.. code-block:: bash
+
+    DB_NAME=skypilot-db
+    DB_INSTANCE_NAME=cloud-sql-skypilot-instance
+    gcloud sql databases create ${DB_NAME} --instance ${DB_INSTANCE_NAME}
+
+Step 3: Create a service account to give access to the cloud SQL instance
+-------------------------------------------------------------------------
+
+In this step, a service account is created to give the API server access to the Cloud SQL instance.
+
+- Go to the `Service Accounts <https://console.cloud.google.com/iam-admin/serviceaccounts>`_ page of the ``IAM and Admin`` console
+- Click on "Create Service Account"
+- Set the service account name and ID to ``skypilot-cloud-sql-access``
+- Click on "Create and Continue" to move to the Permissions page.
+- Click on "Select a member" and add ``Cloud SQL Client`` as the role.
+- Click on "Continue", then "Done" to create the service account.
+
+Step 4: Add in IAM policy binding to the service account
+--------------------------------------------------------
+
+In this step, an IAM policy binding is created on the GCP service account to bind it to the GKE service account.
+
+This step gives the GKE service account access to the GCP service account.
+
+.. code-block:: bash
+
+    NAMESPACE=skypilot
+    GCP_PROJECT_ID=<your gcp project id>
+    GCP_SERVICE_ACCOUNT=skypilot-cloud-sql-access
+    GKE_SERVICE_ACCOUNT=skypilot-api-sa
+    gcloud iam service-accounts add-iam-policy-binding \
+        --role="roles/iam.workloadIdentityUser" \
+        --member="serviceAccount:${GCP_PROJECT_ID}.svc.id.goog[${NAMESPACE}/${GKE_SERVICE_ACCOUNT}]" \
+        ${GCP_SERVICE_ACCOUNT}@${GCP_PROJECT_ID}.iam.gserviceaccount.com
+
+Step 5: Set up the GKE cluster to use the GCP service account
+-------------------------------------------------------------
+
+In this step, the GKE cluster is configured so the API server can use the GCP service account.
+
+
+.. code-block:: bash
+
+    NAMESPACE=skypilot
+    kubectl create namespace ${NAMESPACE}
+
+.. code-block:: bash
+
+    GCP_PROJECT_ID=<your gcp project id>
+    GCP_SERVICE_ACCOUNT=skypilot-cloud-sql-access
+    gcloud iam service-accounts keys create gcp-key.json \
+         --iam-account=${GCP_SERVICE_ACCOUNT}@${GCP_PROJECT_ID}.iam.gserviceaccount.com \
+         --project=${GCP_PROJECT_ID}
+
+.. code-block:: bash
+
+    NAMESPACE=skypilot
+    kubectl create secret generic cloud-sql-credentials \
+        --from-file=service-account-key.json=gcp-key.json -n ${NAMESPACE}
+
+
+Step 6: Deploy the SkyPilot API server
+--------------------------------------
+
+Use the following values.yaml to deploy the SkyPilot API server.
+
+
+``values.yaml``:
+
+.. code-block:: yaml
+
+    apiService:
+      extraVolumes:
+      - name: cloud-sql-credentials
+        secret:
+          secretName: cloud-sql-credentials
+
+      config: |
+        db: postgresql://$DB_USER:$DB_PASSWORD@localhost/$DB_NAME
```

Comment:
can values take env var? If not, we may need to add TODOs to ask people to manually fill in the fields. Also, in that case, we should avoid using env var format here, as people may think it can be taken from env var, maybe using something like `<DB_USER>:<DB_PASSWORD>`

Also, for our user, IIRC, there is a constraint for not being able to specify plain text in the manifest/values. How can they do such with secrets?